src/monodroid/jni/generate-pinvoke-tables.cc

```diff
@@ -0,0 +1,859 @@
+//
+// To build and run this utility run (on Linux or macOS):
+//
+//   ../../../build-tools/scripts/generate-pinvoke-tables.sh
+//
+// A reasonable C++20 compiler is required (g++ 10+, clang 11+, on mac it may require XCode 12.5 or newer)
+//
+// Whenever a new p/invoke (or entire new shared libary which is part of dotnet distribution) is added, try to keep the
+// entries sorted alphabetically.  This is not required by the generator but easier to examine by humans.
+//
+// If a new library is added, please remember to generate a hash of its name and update pinvoke-override-api.cc
+//
+#include <algorithm>
+#include <cerrno>
+#include <cstring>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <iomanip>
+#include <memory>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "xxhash.hh"
+
+namespace fs = std::filesystem;
+using namespace xamarin::android;
+
+const std::vector<std::string> internal_pinvoke_names = {
+	"create_public_directory",
```

Comment:
This feels like a "better late than never" question, but… Why is `create_public_directory` exported?

It's not P/Invoked, afaict, so why is it exported?  What's using it?

Reply:
I think we should leave the review of exported APIs to some later date. I have no idea what uses this function, but there might be something somewhere that does...